xmpp/transports.py

```diff
@@ -277,10 +277,10 @@ def connect(self,dupe=None):
             connection to the target server. Returns non-empty sting on success. """
         if not TCPsocket.connect(self,(self._proxy['host'],self._proxy['port'])): return
         self.DEBUG("Proxy server contacted, performing authentification",'start')
-        connector = ['CONNECT %s:%s HTTP/1.0'%self._server,
+        connector = ['CONNECT %s:%s HTTP/1.0'%dupe,
```

Comment:
Instead of changing self._server to dupe, why not rename dupe to server and add "if not server: server=self._server" like in TCPsocket.connect.  That should continue to support passing server to the constructor.